### PR TITLE
Replace Redux with React Context components

### PR DESCRIPTION
**Pull request by mrtnzlml**
Please read this: https://reactjs.org/docs/context.html

### CODE REVIEW

File: app/hotels/src/allHotels/AllHotelsSearch.js [hunk 1]
```diff
@@ -14,21 +13,23 @@ import type { Coordinates } from '../CoordinatesType';
...
-type Props = {|
-  search: SearchParams,
+type PropsWithContext = {
+  // external:
   openSingleHotel: (searchParams: AvailableHotelSearchInput) => void,
-  onCityIdChange: (cityId: string | null) => void,
   cityId: string,
   coordinates: Coordinates | null,
+  currency: string,
+  // from context
```
Comment (tbergq) [inline app/hotels/src/allHotels/AllHotelsSearch.js]: Why are you not using `...Props` here as done in the other components?

File: app/hotels/src/HotelsFilterContext.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  actions: {|
+    setFilter: OnChangeFilterParams => void,
+  |},
+|};
+
+class Provider extends React.Component<Props, State> {
+  setFilter = (action: OnChangeFilterParams) => {
+    const newState = {
+      ...this.state,
+      filterParams: {
```
Comment (tbergq) [inline app/hotels/src/HotelsFilterContext.js]: Shouldn't you be using functional set state here?

File: app/hotels/src/allHotels/AllHotelsSearch.js (hunk 1 above)
Comment (mrtnzlml) [reply app/hotels/src/allHotels/AllHotelsSearch.js]: I tried but I had to use non-exact props and for some reason, it doesn't work. Can you please check it what's wrong? I wasn't able to fix it this way... 🙈

Comment (tbergq) [reply app/hotels/src/allHotels/AllHotelsSearch.js]: I will have a look and see if I can find out

Comment (tbergq) [reply app/hotels/src/allHotels/AllHotelsSearch.js]: Ok, I can't really find out why this doesn't work. We could merge it and try to make it exact type later in another PR.

Comment (mrtnzlml) [reply app/hotels/src/allHotels/AllHotelsSearch.js]: Should be possible really soon: https://github.com/facebook/flow/issues/2405#issuecomment-379175609

File: app/hotels/src/HotelsFilterContext.js (hunk 2 above)
Comment (mrtnzlml) [reply app/hotels/src/HotelsFilterContext.js]: Yes, fixed - please check it again.

**Pull request merged**